File: day-6.py
from helpers import load_data


def solve(data, marker_size):
	for i in range(marker_size, len(data) + 1):
		if len(set(data[i-marker_size:i])) == marker_size:
			return i


if __name__ == "__main__":
	data = load_data("day-6-input.txt")[0]
	print(solve(data, 4))
	print(solve(data, 14))

File: day-6.hs
import System.IO

main = do
  inputData <- readFile "inputs/day-6-input.txt"
  let inputLine = head (lines inputData)
  print (partOne inputLine)
  -- let inputLine = take 1 lines
  -- print (inputLine)

-- recursion is no bueno for this (problem gets bigger)
partOne :: (Num t) => String -> t
partOne [] = 0
partOne (x:xs) = z + partOne xs
  where z = 1 if 